### PR TITLE
modify GasPrice method fetched from config

### PR DESCRIPTION
**Pull request by zhshy11**
modify GasPrice method fetched from config
more detail : https://github.com/okex/ethermint/issues/346

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/rpc/namespaces/eth/api.go [hunk 1]
```diff
@@ -65,6 +67,19 @@ func NewAPI(
 		panic(err)
 	}
 
+	//gasPrice: to get "minimum-gas-prices" config or to get ethermint.DefaultGasPrice
+	gasPricesStr := viper.GetString(server.FlagMinGasPrices)
+	gasPrices, err := sdk.ParseDecCoins(gasPricesStr)
+	if err != nil {
+		panic(fmt.Sprintf("invalid minimum gas prices: %v", err))
```
Comment (KamiD) [inline app/rpc/namespaces/eth/api.go]: should not panic here

File: app/rpc/namespaces/eth/api.go [hunk 2]
```diff
@@ -65,6 +67,19 @@ func NewAPI(
 		panic(err)
 	}
 
+	//gasPrice: to get "minimum-gas-prices" config or to get ethermint.DefaultGasPrice
```
Comment (KamiD) [inline app/rpc/namespaces/eth/api.go]: package this code segment to a single function

File: app/rpc/namespaces/eth/api.go [hunk 3]
```diff
@@ -65,6 +67,19 @@ func NewAPI(
...
+	gasPricesStr := viper.GetString(server.FlagMinGasPrices)
```
Comment (KamiD) [inline app/rpc/namespaces/eth/api.go]: is `viper.GetString` always success? what's the default return value

File: app/rpc/namespaces/eth/api.go [hunk 4]
```diff
@@ -65,6 +67,19 @@ func NewAPI(
...
+	gasPrices, err := sdk.ParseDecCoins(gasPricesStr)
+	if err != nil {
+		panic(fmt.Sprintf("invalid minimum gas prices: %v", err))
+	}
+	var gasPrice *hexutil.Big
+	if gasPrices == nil || len(gasPrices) == 0 {
+		gasPrice = (*hexutil.Big)(new(big.Int).SetUint64(ethermint.DefaultGasPrice * 1000000000))
```
Comment (KamiD) [inline app/rpc/namespaces/eth/api.go]: don't use magic number such as 1000000000

**Review by KamiD — CHANGES_REQUESTED** [review]

File: app/rpc/namespaces/eth/api.go (hunk 3 above)
Comment (zhshy11) [reply app/rpc/namespaces/eth/api.go]: Empty string

**Pull request closed without merging**